### DISCUSSION TITLE
Meldung obwohl im DP alles ok aussieht?

### DISCUSSION THREAD

**Discussion by mcuiobroker** [Zefau/ioBroker.jarvis#2565] [General]
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/705cc71d-9a1a-4d1f-96f8-4460cc68ab8c)

![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/7d6c94ea-8e27-4efa-bfc5-044f52e42b87)
```
{
  "'{0_userdata.0.jarvis.test.text3}'='Offline'": "mdi:pump blink",
  "true": "mdi:pump",
  "false": "mdi:pump"
}
```
Icon Stil
```
{
  "'{0_userdata.0.jarvis.test.text3}' ='Offline'": {
    "color": "#FF0"
  },
  "true": {
    "color": "#0F0"
  },
  "false": {
    "color": "#F00"
  }
}
```

**Comment by Bernd9000** [comment]
Versuch mal Bitte den Namen vom Datenpunkt-Schlüssel "Zuhause" in z.B. "Zuhause2" umzubenennen.
Bei mir durfte es nicht "power" sein.

[#2534](https://github.com/Zefau/ioBroker.jarvis/issues/2534#issuecomment-1904560921)

**Comment by mcuiobroker** [comment]
Bei mir ist noch interessanter
Neues Widget auf der ersten Seite
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/2e673613-5317-481e-8431-aafc4dc33879)

![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/e376919d-b7e5-4fd2-9cf9-7923f0bf8ce8)

![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/7cc4e14a-e7c7-4a35-8741-4b6ffb690faa)

Es wird keine Switch angezeigt.
Trotzdem die Meldungen
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/8152eb94-2e18-406a-aee1-2ad08f8daf3c)

Im Popup ist der Switch da
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/014ee75f-a1cd-44a9-9ece-e204f3bdb3fd)

Schalte ich Styles und scripts aus
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/5c8b4095-3e5a-4fad-8c87-9ecc0bff84c2)

**Comment by Bernd9000** [comment]
Vielleicht ein Problem mit dem Unterstrich in "STATUS_ZUHAUSE" ?

**Comment by mcuiobroker** [comment]
Aktualisierte DP `0_userdata.0.Haus.allgemein.status.zuHause `
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/64feb18d-6f81-4965-a660-e19e987b6b14)
evtl der Grossbuchstabe in der ID?

**Comment by mcuiobroker** [comment]
Bekomme beim Schalten auch das
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/11f2872e-3277-4803-9ab8-1eab8d66bb4a)
Obwohl vorhanden
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/9c59dd72-c6e3-4730-a497-e897010c062c)

![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/10ab0e84-c74c-49fe-abe6-8f5acfd1f488)

**Comment by mcuiobroker** [comment]
Alter DP
```
"0d451c75-6bf9-44fe-b661-563c6c44d64d": {
    "name": "Status Zuhause",
    "aliases": "",
    "tags": [],
    "icon": "",
    "label": "",
    "function": "_defaults",
    "states": {
      "Zuhause2": {
        "showState": false,
        "icon": {
          "'{0_userdata.0.jarvis.test.text3}'='Offline'": "mdi:pump blink",
          "true": "mdi:pump",
          "false": "mdi:pump"
        },
        "state": "0_userdata.0.Haus.allgemein.status.zuHause",
        "actionElement": "SwitchAction",
        "iconStyle": {
          "'{0_userdata.0.jarvis.test.text3}' ='Offline'": {
            "color": "#FF0"
          },
          "true": {
            "color": "#0F0"
          },
          "false": {
            "color": "#F00"
          }
        },
        "bodyElement": "LastChangeBody",
        "stateKey": "Zuhause2",
        "SwitchActionConfig": {
          "colorOn": "blue",
          "colorOff": "green"
        },
        "_created": 1705951245565,
        "_updated": 1705951245565
      }
    },
    "options": {
      "suppressPopup": false
    },
    "attributes": {
      "_created": 1604933910385,
      "_updated": 1705951810477
    },
    "id": "0d451c75-6bf9-44fe-b661-563c6c44d64d",
    "revision": "d937c"
  },
```


Neuer DP
``` 
"ee3dae37-41e7-4957-bfcd-47a5dd534a6a": {
    "name": "Test Zuhause",
    "aliases": "",
    "tags": [],
    "icon": "",
    "label": "",
    "function": "_defaults",
    "states": {
      "zuhause": {
        "_created": 1705952246844,
        "stateKey": "zuhause",
        "showState": false,
        "state": "0_userdata.0.Haus.allgemein.status.zuHause",
        "_updated": 1705952585677,
        "icon": {
          "'{0_userdata.0.jarvis.test.text3}'='Offline'": "mdi:pump blink",
          "true": "mdi:pump",
          "false": "mdi:pump"
        },
        "iconStyle": {
          "'{0_userdata.0.jarvis.test.text3}' ='Offline'": {
            "color": "#FF0"
          },
          "true": {
            "color": "#0F0"
          },
          "false": {
            "color": "#F00"
          }
        },
        "bodyElement": "LastChangeBody",
        "actionElement": "SwitchAction",
        "SwitchActionConfig": {
          "colorOn": "blue",
          "colorOff": "green"
        }
      }
    },
    "options": {
      "suppressPopup": false
    },
    "id": "ee3dae37-41e7-4957-bfcd-47a5dd534a6a",
    "attributes": {
      "_created": 1705952231743,
      "_updated": 1705952565940
    },
    "revision": "088ff"
  }
```

![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/651a0791-507c-41fa-880f-e5a0236ad3a0)

Neuer DP oben
Parameter sind identisch vorhanden -> sollte das gleiche Ergebnis haben.

**Comment by mcuiobroker** [comment]
Deswegen gehe ich davon aus, dass irgendwo im Speicher etwas gehalten wird.
Also benötigt man einen Reset-Schalter mit dem man auch diesen internen Kernspeicher löschen kann. Falls solche Problem entstehen.

**Comment by Bernd9000** [comment]
Da wo kein `switch` angezeigt wird fehlte bei mir das Trigger Element in den Widgets bereits in der alten Version.
Warum auch immer das voher nicht gespeichert wurde.
Die alte Version hat dann vermutlich wenn nichts eingetragen ist einfach als standart den switch Action genommen.

Also das hier fehlte bei mir in den betroffenen Widgets (Expert Mode):
"actionElement": "SwitchAction"

Man kann das im Widget nochmal auswählen und dann wird es auch gespeichert.

Erkennbar auch an dem fehlendem Auswahlelement im Widget:

Falsch:
![1](https://github.com/Zefau/ioBroker.jarvis/assets/77005685/c66a0686-cecc-4f3c-88fe-83918d76b344)

Richtig:
![2](https://github.com/Zefau/ioBroker.jarvis/assets/77005685/e5c4eacb-cb69-4ba1-a7bc-6a743a11ee08)

Eigentlich sollte dann ja der Eintrag vom Gerät erfolgen (Rückfallebene), aber da fehlte bei mir auch das "actionElement"
Bzw. die Rückfallebene funktioniert nicht richtig ?

**Comment by mcuiobroker** [comment]
Jetzt per Zufall nachgeschaut.
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/1cc5d64c-f610-4721-83c5-c06904a5f4a2)
~Bewusst nichts geändert.~ Doch was geändert, dass war die Version v3.1.9-beta.2, da wird es also noch angezeigt.
`v3.2.0-alpha.52`
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/24da4d78-3847-4ba6-af90-a3907d6ebef0)

SwitchAction nur referenziert im Layout.
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/003f1a78-3e1e-485a-a98e-a40100aa12de)

**Comment by Zefau** [comment]
Die Probleme hier sollten mit beta-5 alle behoben sein, korrekt?
